pkg/credentialprovider/provider.go

```diff
@@ -77,8 +78,11 @@ func (d *defaultDockerConfigProvider) Enabled() bool {
 	return true
 }
 
-// Provide implements dockerConfigProvider
-func (d *defaultDockerConfigProvider) Provide() DockerConfig {
+// Provide returns docker configuration. Implementations can be blocking - e.g.
+// metadata server unavailable. The image is passed into Provide as context in
+// the event that the implementation depends on information in the image name
+// to return credentials; implementations are safe to ignore the image.
```

Comment:
this is good documentation, I would hoist it up to the interface method. This implies callers of the interface must call Provide() (or LazyProvide()) for every distinct image they deal with

Reply:
what should the provider do with an image it doesn't recognize? what about an malformed image spec?

Reply:
For AWS it is checking if it is a possibly valid ECR path and logging the error as an info(3) and then returning credentialprovider.DockerConfig{}. I figure it would be specific to the cloud provider.